blint/binary.py

```diff
@@ -983,6 +985,27 @@ def parse_go_buildinfo(parsed_obj: lief.Binary) -> (dict[str, dict[str, str]], d
     return deps, formulation
 
 
+def parse_rust_buildinfo(parsed_obj: lief.Binary) -> list:
+    """
+    Parse the rust build info section of binaries that are cargo-auditable to extract rust dependencies
+    Args:
+        parsed_obj (lief.Binary): The parsed object representing the binary.
+
+    Returns:
+        list: List representing the dependencies.
+    """
+    deps = {}
+    audit_data_section = next(filter(lambda section: section.name == ".dep-v0", parsed_obj.sections))
```

Comment:
Looks like filter function is returning empty for non-rust binaries so the next call is failing. Perhaps we need to use `has_section`?

```
if parsed_obj.has_section(".dep-v0"):
```

Reply:
Good call. When I try your suggestion, I am getting an error: `'Binary' object has no attribute 'has_section'`

I will commit an alternative approach to address this issue

Reply:
Try like this.
```
with contextlib.suppress(AttributeError, TypeError):
    if parsed_obj.has_section(".dep-v0"):
```

Reply:
@cerrussell any ideas why we might get attribute error for has_section?